src/main/java/org/springframework/cloud/deployer/spi/kubernetes/AbstractKubernetesDeployer.java

```diff
@@ -183,9 +182,18 @@ PodSpec createPodSpec(AppDeploymentRequest appDeploymentRequest) {
 
 		ContainerConfiguration containerConfiguration = new ContainerConfiguration(appId, appDeploymentRequest)
 				.withProbeCredentialsSecret(getProbeCredentialsSecret(deploymentProperties))
-				.withExternalPort(externalPort)
 				.withHostNetwork(hostNetwork);
 
+		if (KubernetesAppDeployer.class.isAssignableFrom(this.getClass())) {
+			containerConfiguration.withExternalPort(getExternalPort(appDeploymentRequest));
+		}
+		else {
```

Comment:
since before tasks were always setting null for port, is this needed?

Reply:
Good point. We should only set the port for `K8sAppDeployer` to be in sync with the previous versions.